modules/auxiliary/gather/safari_file_url_navigation.rb

```diff
@@ -322,7 +322,7 @@ def cleanup
     super
 
     # Kill FTP
-    stop_service
+    cleanup_service
 
     # clear my resource, deregister ref, stop/close the HTTP socket
     begin
```

Comment:
Looks like this has explicit calls to `@http_service.stop` and `@http_server.close`, I feel like that's something that we can remove now? 👀

Reply:
Running this module as well as a http handler together on the same ip/port:
```
msf6 auxiliary(gather/safari_file_url_navigation) > jobs

Jobs
====

  Id  Name                                          Payload                           Payload opts
  --  ----                                          -------                           ------------
  6   Auxiliary: gather/safari_file_url_navigation
  7   Exploit: multi/handler                        osx/x64/meterpreter_reverse_http  http://192.168.123.1:8080/base_url
```

Getting sessions:
```
msf6 auxiliary(gather/safari_file_url_navigation) > sessions 

Active sessions
===============

  Id  Name  Type                 Information               Connection
  --  ----  ----                 -----------               ----------
  2         meterpreter x64/osx  a @ 192.168.1.239  192.168.123.1:8080 -> 127.0.0.1  ()
  3         meterpreter x64/osx  a @ 192.168.1.239  192.168.123.1:8080 -> 127.0.0.1  ()


```

Killing the safari file module:
```
msf6 auxiliary(gather/safari_file_url_navigation) > jobs -k 6
[*] Stopping the following job(s): 6
[*] Stopping job 6

[*] Server stopped.
```

Verifying jobs:
```
msf6 auxiliary(gather/safari_file_url_navigation) > jobs

Jobs
====

  Id  Name                    Payload                           Payload opts
  --  ----                    -------                           ------------
  7   Exploit: multi/handler  osx/x64/meterpreter_reverse_http  http://192.168.123.1:8080/base_url
```

Interacting with a previously opened session is now dead:

```
msf6 auxiliary(gather/safari_file_url_navigation) > sessions -i -1 -C dir
[*] Running 'dir' on meterpreter session 3 ()


[-] Error running command dir: Rex::TimeoutError Operation timed out.
```

Reply:
Thanks - I've removed the `stop` and `close`, and let `deref` do its thing.